Result of `nixpkgs-review pr 107009` run on x86_64-darwin [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package failed to build:</summary>
  <ul>
    <li>pappl</li>
  </ul>
</details>

```
3 warnings generated.
Compiling job-accessors.c...
In file included from device-usb.c:15:
In file included from ./device-private.h:17:
./base-private.h:43:13: warning: 'strlcpy' macro redefined [-Wmacro-redefined]
#    define strlcpy(dst,src,dstsize) _pappl_strlcpy(dst,src,dstsize)
            ^
/nix/store/r913a0b23i21r8y4x8kgx024gms9lvx9-Libsystem-osx-10.12.6/include/secure/_string.h:104:9: note: previous definition is here
#define strlcpy(dest, src, len)                                 \
        ^
In file included from dnssd.c:15:
In file included from ./pappl-private.h:19:
In file included from ./dnssd-private.h:18:
./base-private.h:43:13: warning: 'strlcpy' macro redefined [-Wmacro-redefined]
#    define strlcpy(dst,src,dstsize) _pappl_strlcpy(dst,src,dstsize)
            ^
/nix/store/r913a0b23i21r8y4x8kgx024gms9lvx9-Libsystem-osx-10.12.6/include/secure/_string.h:104:9: note: previous definition is here
#define strlcpy(dest, src, len)                                 \
        ^
dnssd.c:267:10: error: use of undeclared identifier 'kDNSServiceErr_DefunctConnection'; did you mean 'kDNSServiceFlagsShareConnection'?
    case kDNSServiceErr_DefunctConnection :
         ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
         kDNSServiceFlagsShareConnection
/nix/store/r913a0b23i21r8y4x8kgx024gms9lvx9-Libsystem-osx-10.12.6/include/dns_sd.h:277:5: note: 'kDNSServiceFlagsShareConnection' declared here
    kDNSServiceFlagsShareConnection     = 0x4000,
    ^
1 warning and 1 error generated.
make[1]: *** [../Makedefs:72: dnssd.o] Error 1
make[1]: *** Waiting for unfinished jobs....
1 warning generated.
In file included from job-accessors.c:14:
In file included from ./pappl-private.h:19:
In file included from ./dnssd-private.h:18:
./base-private.h:43:13: warning: 'strlcpy' macro redefined [-Wmacro-redefined]
#    define strlcpy(dst,src,dstsize) _pappl_strlcpy(dst,src,dstsize)
            ^
/nix/store/r913a0b23i21r8y4x8kgx024gms9lvx9-Libsystem-osx-10.12.6/include/secure/_string.h:104:9: note: previous definition is here
#define strlcpy(dest, src, len)                                 \
        ^
1 warning generated.
1 warning generated.
make[1]: Leaving directory '/private/tmp/nix-build-pappl-1.0.0.drv-0/source/pappl'
make: *** [Makefile:28: all] Error 1
```

--- a/pkgs/applications/printing/pappl/default.nix
+++ b/pkgs/applications/printing/pappl/default.nix
@@ -1,0 +1,62 @@
+{ lib, stdenv, fetchFromGitHub
+, avahi
+, cups
+, gnutls
+, libjpeg
+, libpng
+, libusb1
+, pkg-config
+, withPAMSupport ? true, pam
+, zlib
+}:
+
+stdenv.mkDerivation rec {
+  pname = "pappl";
+  version = "1.0.0";
+
+  src = fetchFromGitHub {
+    owner = "michaelrsweet";
+    repo = pname;
+    rev = "v${version}";
+    sha256 = "1cg06v8hxska0hnybnmfda1v4h3ifjir24nx2iqx80kb6jq0hayb";
+  };
+
+  outputs = [ "out" "dev" ];
+
+  nativeBuildInputs = [
+    pkg-config
+  ];
+
+  buildInputs = [
+    cups
+    libjpeg
+    libpng
+    libusb1
+    zlib
+  ] ++ lib.optionals (!stdenv.isDarwin) [
+    # upstream mentions these are not needed for Mac
+    # see: https://github.com/michaelrsweet/pappl#requirements
+    avahi
+    gnutls
+  ] ++ lib.optionals withPAMSupport [
+    pam
+  ];
+
+  # testing requires some networking
+  # doCheck = true;
+
+  doInstallCheck = true;
+  installCheckPhase = ''
+    $out/bin/pappl-makeresheader --help
+  '';
+
+  enableParallelBuilding = true;
+
+  meta = with lib; {
+    description = "C-based framework/library for developing CUPS Printer Applications";
+    homepage = "https://github.com/michaelrsweet/pappl";
+    license = licenses.asl20;
+    platforms = platforms.linux; # should also work for darwin, but requires additional work
+    maintainers = with maintainers; [ jonringer ];
+  };
+}

--- a/pkgs/top-level/all-packages.nix
+++ b/pkgs/top-level/all-packages.nix
@@ -6408,6 +6408,8 @@ in
 
   papertrail = callPackage ../tools/text/papertrail { };
 
+  pappl = callPackage ../applications/printing/pappl { };
+
   par2cmdline = callPackage ../tools/networking/par2cmdline { };
 
   parallel = callPackage ../tools/misc/parallel { };

